src/coreComponents/constitutive/fluid/multifluid/MultiFluidSelector.hpp

```diff
@@ -79,6 +79,111 @@ void constitutiveUpdatePassThru( MultiFluidBase & fluid,
                                >::execute( fluid, std::forward< LAMBDA >( lambda ) );
 }
 
+namespace detail
+{
+
+// Lists the possible numbers of components supported by a fluid type
+// Most of the fluid types support only 2 components so this is the default
+template< typename FLUID_TYPE >
+struct Components
+{
+  using type = camp::int_seq< integer, 2 >;
+};
+
+// Dead oil fluid model supports 2 or 3 components
+template<>
+struct Components< DeadOilFluid >
+{
+  using type = camp::int_seq< integer, 2, 3 >;
+};
+
+// Blackoil fluid model supports 3 components
+template<>
+struct Components< BlackOilFluid >
+{
+  using type = camp::int_seq< integer, 3 >;
+};
+
+// Compositional fluid models support anything from 2 to 5 components
+template<>
+struct Components< CompositionalMultiphaseFluidPVTPackage >
+{
+  using type = camp::int_seq< integer, 2, 3, 4, 5 >;
+};
+template<>
+struct Components< CompositionalTwoPhaseLohrenzBrayClarkViscosity >
+{
+  using type = camp::int_seq< integer, 2, 3, 4, 5 >;
+};
+template<>
+struct Components< CompositionalTwoPhaseConstantViscosity >
+{
+  using type = camp::int_seq< integer, 2, 3, 4, 5 >;
```

Comment:
I think that you can do something like this to avoid specifying each number:

````c++
template<Integer Begin, integer End, integer... Is>
struct create_integer_sequence_impl
    :  create_integer_sequence_impl<Begin, End - 1, End - 1, Is...> {};

template<Integer Begin, Integer... Is>
struct create_integer_sequence_impl<Begin, Begin, Is...>
{
    using type = camp::int_seq<int, Is...>;
};
````
and then use it like:

````c++
template<>
struct Components< CompositionalTwoPhaseConstantViscosity >
{
using type = typename create_integer_sequence_impl<2, 5>::type;
}
````

and maybe I would define the value in the max and min value in the fluid model itself and have something like:

````c++
template<typename FLUID_TYPE>
struct Components< >
{
using type = typename create_integer_sequence_impl<FLUID_TYPE::min_n_components, FLUID_TYPE::max_n_components>::type;
}
````
@wrtobin thoughts?

Reply:
IIRC camp has builtin support for range generation... but looking at it it seems like they only support [0...N) generation.
So yeah we should be able to avoid explicitly listing all numbers unless there is a compelling reason to do so.

I think its typically logical to add a third component for the step size which defaults to 1, so we can e.g. count by 2's etc, but that is just as a general mechanism.

Reply:
if we make it even more generic it could probably even go in `common` or in `datatypes`. Honestly, it should probably live directly in `camp`. I also had a look in `camp` and I didn't see exactly what we need. 

In any case I don't see a real reason for explicitly listing all numbers, so I think we should do something like what I wrote above but @dkachuma feel free to also the step size and move it somewhere else.

Reply:
I particularly like the idea of min/max component with each fluid. Let me see how I can fit this in.